### ISSUE TITLE
AudioVideoPlaybackBot - Container name 'aus1-avb-stageartifacts-' is invalid

### ISSUE THREAD

**Issue by osamabinsaleem** [microsoftgraph/microsoft-graph-comms-samples#688]
When I run the following command to deploy the infrastructure I get multiple errors. 
Command: 
```
.\deploy.ps1 -orgName TBA -Location uksouth -RunDeployment
```

Errors: 
```
Container name 'aus1-avb-stageartifacts-' is invalid. Valid names start and end with a lower case letter or a number and has in
     | between a lower case letter, number or dash with no consecutive dashes and is 3 through 63 characters long.
Set-AzStorageBlobContent:


 327 |  … $global:r = New-AzDeployment @Common @TemplateArgs @OptionalParameter …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot retrieve the dynamic parameters for the cmdlet. Cannot find Bicep. Please add Bicep to your PATH or visit
     | https://aka.ms/bicep-install to install Bicep.

```

Also I've a confusion. My cloud service is named as 'team-bot-alfred'. So, my DNS would be 'team-bot-alfred.cloudapp.net', right?

**Comment by InDieTasten** [comment]
You have to define an environment variable called `USERNAME` and set that to something that ends with an alphanumeric character.

You can do this by running something like

    $env:USERNAME = 'osamabinsaleem'

The script is trying to create a storage account, and the environment variable is being used to generate as the last part of that name. The naming of storage accounts is very strict. The error message from your initial post describes what the fully generated name needs to adhere to.

Reference to the part of the script utilizing the environment variable:
https://github.com/microsoftgraph/microsoft-graph-comms-samples/blob/87aa1b2fca82b8373bc080d4e8552f02448adfa9/Samples/V1.0Samples/LocalMediaSamples/AudioVideoPlaybackBot/ADF/release-az/Start-AzDeploy.ps1#L129

**Comment by InDieTasten** [comment]
> Also I've a confusion. My cloud service is named as 'team-bot-alfred'. So, my DNS would be 'team-bot-alfred.cloudapp.net', right?

I don't think so. As far as I know, the fqdn will end up being `team-bot-alfred.uksouth.cloudapp.azure.com`, at least that's how Azure Public IP domains are generated. Not too sure about classic cloudapps though :)
